src/mbgl/renderer/layers/render_location_component_layer.cpp

```diff
@@ -0,0 +1,895 @@
+#include <array>
+#include <mapbox/cheap_ruler.hpp>
+#include <mbgl/gfx/backend_scope.hpp>
+#include <mbgl/gfx/renderer_backend.hpp>
+#include <mbgl/gl/context.hpp>
+#include <mbgl/gl/renderable_resource.hpp>
+#include <mbgl/map/transform_state.hpp>
+#include <mbgl/platform/gl_functions.hpp>
+#include <mbgl/renderer/bucket.hpp>
+#include <mbgl/renderer/layers/render_location_component_layer.hpp>
+#include <mbgl/renderer/paint_parameters.hpp>
+#include <mbgl/style/layers/location_component_layer.hpp>
+#include <mbgl/style/layers/location_component_layer_impl.hpp>
+#include <mbgl/style/layers/location_component_layer_properties.hpp>
+#include <mbgl/util/mat4.hpp>
+
+#include <mapbox/eternal.hpp>
+#include <mbgl/gl/context.hpp>
+#include <mbgl/gl/defines.hpp>
+#include <mbgl/gl/texture.hpp>
+#include <mbgl/gl/texture_resource.hpp>
+#include <mbgl/gl/types.hpp>
+#include <mbgl/platform/gl_functions.hpp>
+#include <mbgl/renderer/image_manager.hpp>
+
+using namespace mbgl::platform;
+namespace mbgl {
+namespace style {
+struct LocationComponentRenderParameters {
```

Comment:
Should be removed, PaintParameters + paint / layout properties should be enough to calculate all uniforms and prepare data to draw layer.

Reply:
It's also used to diff and update only what changed.